File: CURIOSITY/bootcamp-research/_Aly-Lei/Quester/app/models/avatar.py
from .db import db, c


class Avatar(db.Model):
    __tablename__ = "avatars"

    id = c(db.Integer, primary_key=True)
    user_id = c(db.Integer, db.ForeignKey('users.id'), nullable=False)
    prebuilt = c(db.String(50), nullable=False)
    hair = c(db.String(50), nullable=True)
    face = c(db.String(50), nullable=True)
    body = c(db.String(50), nullable=True)
    helmet = c(db.String(50), nullable=True)
    top = c(db.String(50), nullable=True)
    bottom = c(db.String(50), nullable=True)

    def to_dict(self):
        return {
            "prebuilt": self.prebuilt,
            "hair": self.hair,
            "face": self.face,
            "body": self.body,
        }

File: CURIOSITY/bootcamp-research/_Aly-Lei/Quester/react-app/src/User/Avatar.js
import React from "react";
import { useSelector } from "react-redux";

const Avatar = () => {
  const avatar = useSelector((state) => state.avatar.avatar);

  return (
    <>
      <div>
        <h2>Avatar</h2>
        Hair: {avatar.hair}, Face: {avatar.face}, Body: {avatar.body}
      </div>
    </>
  );
};

export default Avatar;
